src/Components/Server/src/ComponentHub.cs

```diff
@@ -110,6 +110,18 @@ public string StartCircuit(string baseUri, string uri)
                 return null;
             }
 
+            if (baseUri == null ||
+                uri == null ||
+                !Uri.IsWellFormedUriString(baseUri, UriKind.Absolute) ||
+                !Uri.IsWellFormedUriString(uri, UriKind.Absolute))
+            {
```

Comment:
What would prevent me from sending StartCircuit www.bing.com and www.bing.com/something. Shouldn't we check that these uris match the app up to the PathBase?

Reply:
Unfortunately that's not straightforward to do. We don't have a requirement that the base URI match the websocket request. We also don't really have a list of "known valid base paths" to check against.